File: arvoreDecisao.js
const DTC = require('decision-tree-classifier');
const iris = require('ml-dataset-iris');

const data = iris.getNumbers();
const target = iris.getLabels();
const [X_train, X_test, y_train, y_test] = splitData(data, target);

const dtc = new DTC({
  gainFunction: 'gini',
  maxDepth: 3
});
dtc.train(X_train, y_train);

const y_pred = dtc.predict(X_test);
const accuracy = getAccuracy(y_test, y_pred);
console.log(`Acurácia: ${accuracy}`);

File: arvoreDecisao.py
from sklearn.datasets import load_iris
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

iris = load_iris()
X = iris.data
y = iris.target

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

clf = DecisionTreeClassifier(random_state=42)
clf.fit(X_train, y_train)

y_pred = clf.predict(X_test)
print('Acurácia:', accuracy_score(y_test, y_pred))
